src/main/java/mekanism/common/content/gear/mekatool/ModuleVeinMiningUnit.java

```diff
@@ -49,30 +51,36 @@ public int getExcavationRange() {
         return excavationRange.get().getRange();
     }
 
-    public static Set<BlockPos> findPositions(BlockState state, BlockPos location, Level world, int maxRange) {
-        Set<BlockPos> found = new LinkedHashSet<>();
-        Set<BlockPos> openSet = new LinkedHashSet<>();
-        openSet.add(location);
-        Block startBlock = state.getBlock();
-        int maxCount = MekanismConfig.gear.disassemblerMiningCount.get() - 1;
-        while (!openSet.isEmpty()) {
-            BlockPos blockPos = openSet.iterator().next();
-            found.add(blockPos);
+    public static Map<BlockPos, BlockState> findPositions(Level world, Map<BlockPos, BlockState> initial, boolean extended, int maxRange) {
+        int maxVein = MekanismConfig.gear.disassemblerMiningCount.get();
+        int maxCount = (int) (initial.size() + maxVein * initial.values().stream().map(bs->bs.getBlock()).distinct().count());
```

Comment:
Previously it was 1 (target) + maxVein - 1
Now it is initialCount + typeCount * maxVein + 1
which for the default not blasting case is just 1 + maxVein + 1
The +1 at the end is from the stop condition being found > max instead of found >= max.
I think I'll change that, but it's not technically vein mining the initial set, so I think that would make sense then.